lib/layout.js

```diff
@@ -615,6 +616,20 @@ export default class PodiumLayout {
                 res.podiumSend = (data, ...args) =>
                     res.send(this.render(incoming, data, ...args));
 
+                // we add a new podiumStream method that sets up streaming and returns the stream
+                // object for the developer to work with
+                res.podiumStream = (...args) => {
+                    const responseStream = new ResponseStream();
+                    // pipe the readable response stream into the express writeable res stream
+                    // to set up streaming
+                    responseStream.pipe(res);
+                    // html template changed to accept body as either a string as before
+                    // or as in this case, a response stream object.
+                    this.#view(incoming, responseStream, ...args);
+                    // return the stream for the developer to work with
+                    return responseStream;
+                };
```

Comment:
I went with a new function on res here. res.podiumStream which can be passed additional arguments for html template and returns a stream object to work with after setting up streaming internally.

Reply:
with this approach, we can stay non breaking and any page that wants to implement streaming just needs to ensure that all podlets on the page are updated to a a version of @podium/podlet that supports early hints.

Reply:
existing templates continue to work as before since we just inject a token and use it to split the template into 2 pieces. We then immediately send the first (header) and hand control to the developer to do streaming. Once the developer is finished and hands control back, we send the footer and terminate the stream.

Reply:
> Worth noting that there will need to be a slightly different implementation for Fastify than Express

The difference are under the hood or also in the public API surface?

Reply:
The only difference would be in how we pipe the streams in the fastify plugin, api surface should be the same for the end user